server/storage/mvcc/index.go

```diff
@@ -67,9 +67,13 @@ func (ti *treeIndex) Put(key []byte, rev revision) {
 }
 
 func (ti *treeIndex) Get(key []byte, atRev int64) (modified, created revision, ver int64, err error) {
-	keyi := &keyIndex{key: key}
 	ti.RLock()
 	defer ti.RUnlock()
+	return ti.get(key, atRev)
+}
+
+func (ti *treeIndex) get(key []byte, atRev int64) (modified, created revision, ver int64, err error) {
```

Comment:
@SimFG  I have few comments: 
1) Why we need to modify `Get`?. The new `get` method can be mistakenly used in future without setting lock..we can add some doc for `get` but even then why not just use `Get` as it's being used currently and we don't worry about every caller func to `get` set lock. 
2) I would prefer to have PR stick to what's in the description, that we are getting rid of few redundant lines of code by using `visit`. 
3) Can't we just use`visit` in `RangeSince` by only change in this PR to call `visit` and remove the two lines of lock call similar to how `visit` is used in the `Revisions`?  Thanks!

Reply:
1. Because if doing like this, the lock will be only used in the exposed method, like `Put`, `KeyIndex`. And the internal method will be without the lock which makes these functions clearer.
2. Yes, I agree.
3. I have thought of doing this. When using `visit` to replace `AscendGreaterOrEqual`, the lock in this method should be moved into the `if` inside, which is strange. If removing the lock and using `ti.Get` like `Revisions`, the `keyi.since` isn't safe without the lock.